pytest_html/plugin.py

```diff
@@ -551,30 +583,12 @@ def generate_summary_item(self):
         return unicode_doc.decode("utf-8")
 
     def _generate_environment(self, config):
-        if not hasattr(config, "_metadata") or config._metadata is None:
-            return []
-
-        metadata = config._metadata
-        environment = [html.h2("Environment")]
-        rows = []
-
-        keys = [k for k in metadata.keys()]
-        if not isinstance(metadata, OrderedDict):
-            keys.sort()
-
-        for key in keys:
-            value = metadata[key]
-            if isinstance(value, str) and value.startswith("http"):
-                value = html.a(value, href=value, target="_blank")
-            elif isinstance(value, (list, tuple, set)):
-                value = ", ".join(str(i) for i in sorted(map(str, value)))
-            elif isinstance(value, dict):
-                sorted_dict = {k: value[k] for k in sorted(value)}
-                value = json.dumps(sorted_dict)
-            raw_value_string = raw(str(value))
-            rows.append(html.tr(html.td(key), html.td(raw_value_string)))
-
-        environment.append(html.table(rows, id="environment"))
+        environment_table = self.EnvironmentTable(config).environment_table
+        if environment_table:
+            environment = [html.h2("Environment")]
```

Comment:
This could be a title attribute on the EnvironmentTable class.

Reply:
Yes will be done on the next version